File: 300.600/387.FirstUniqueCharacterInAString.cpp
#include <iostream>
#include <string>
#include <array>
#include <queue>
using namespace std;

class FirstUniqueCharacterInAString{
public:
    int firstUniqChar(string s) {
        array<int,26> used{};
        for(char c : s){
            used[c-'a']+=1;
        }
        for(int i = 0; i<s.size();++i){
            if(used[s[i]-'a'] == 1)
                return i;
        }
        return -1;
    }
};

int main(int argc, char const *argv[]){
    FirstUniqueCharacterInAString cls;
    cout << cls.firstUniqChar("leetcode") << " -> 0" << endl;
    cout << cls.firstUniqChar("loveleetcode") << " -> 2" << endl;
    return 0;
}

File: go/300.500/387.FirstUniqueCharacterInAString.go
package main

import "fmt"

func firstUniqChar(s string) int {
	used := make([]int, 26)
	byts := []byte(s)
	for _, c := range byts {
		used[c-'a']++
	}
	for i, c := range byts {
		if used[c-'a'] == 1 {
			return i
		}
	}
	return -1
}

func main() {
	fmt.Println(firstUniqChar("leetcode"), 0)
	fmt.Println(firstUniqChar("loveleetcode"), 2)
}
